assets/js/filter.js

```diff
@@ -0,0 +1,174 @@
+$(function () {
+    // contentApiKey is instance-dependent; replace this with a new key if
+    // testing on a local instance
+    // new SearchinGhostEasy({ contentApiKey: '3ecdb68993794bc0157e776e5d' });
+
+    // Filter feature
+    var primaryFilters = "";
+    var internalFilters = "";
+
+    function getFilterOptions(target) {
+        // Retrieve select value from filter and manipulate the values for filtering
+        var filterOptions = $(target).val() ? $(target).val() : [];
+        // Check if target is an internal tag filter
+        var internalTag = $(target).hasClass("internal-tag");
+        // Set respective prefix
+        var tagPrefix = internalTag ? ".tag-hash-" : ".tag-";
+        // Initialise a new array for holding raw values to be written to localstorage
+        var rawFilter = new Array();
+        // Loop through each item in the filter
+        $.each(filterOptions, function (index, value) {
+            // Push the raw value for localstorage
+            rawFilter.push(value);
+            filterOptions[index] = tagPrefix + slugRegex(value);
+        });
+        if (internalTag) {
+            // Write values for internal tags
+            internalFilters = filterOptions;
+            // Write internal filters to localstorage for persistance across pages
+            localStorage.setItem("internalFilters", rawFilter.join("|"));
+        } else {
+            // Write values for primary tags
+            primaryFilters = filterOptions;
+            // Write filters to localstorage for persistance across pages
+            localStorage.setItem("primaryFilters", rawFilter.join("|"));
+        }
+    }
+    function slugRegex(value) {
+        // ! Potentially breaking regex
+        // Basic slug generator: replace whitespaces with dashes, then strip punctuation
+        return value
+            .replace(/\s+/g, "-")
+            .replace(/[.,'!?]/, "")
+            .toLowerCase();
+    }
+    function filterPosts() {
+        // Get filter options for both filter categories
+        getFilterOptions($("#primary-filter"));
+        getFilterOptions($("#internal-filter"));
+        var filters = [];
+        if (primaryFilters.length >= 1 && internalFilters.length >= 1) {
+            // Combine filters for an AND relationship filter
+            $.each(primaryFilters, function (indexP, valueP) {
+                $.each(internalFilters, function (indexI, valueI) {
+                    filters.push(valueP + valueI);
+                });
+            });
+        } else {
+            // Get filters
+            filters = primaryFilters.concat(internalFilters);
+        }
+        // Hide all posts by default
+        $(".gh-postfeed")
+            .children()
+            .each(function () {
+                $(this).addClass("d-none");
+            });
```

Comment:
I think this should be shifted into the `if` block below. Otherwise, even if there are no filters, all posts will be hidden by default also.